barretenberg/cpp/pil/vm2/poseidon2_hash.pil

```diff
@@ -67,21 +67,20 @@ namespace poseidon2_hash;
     // The permutation output values are represented by b_0, b_1, b_2, b_3
     // This most definitely could be simplified to a lower degree check
     // The next perm input is constrained to be the previous perm output + the new values to be hashed if we
-    // are not at the start or the end of hashing
-    pol NEXT_INPUT_IS_PREV_OUTPUT_SEL = (1 - LATCH_CONDITION) * (1 - start);
```

Comment:
This was a bug 😨 . It was too relaxed, it's changed to only be the `(1 - LATCH_CONDITION)`